Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -1207,37 +1207,68 @@ static void CANDataRegWrite ( uint8 * pui8Data, uint32 * pui32Register , uint8 u
  - This function performs the polling of CAN controller mode transitions.
  - it implements the polling of CAN status register flags to detect transition of CAN Controller state.
  ----------------------------------------------------------------------*/
-void Can_MainFunction_Mode(void) {
-	uint8 ControllerIndex;
-	/* initialize base address to controller 0*/
-	/* still need to enter a loop to do the work for controller 0 and 1. not controller 0 only as now */
-	uint32 ui32Base = Global_Config->CanConfigSetRef->CanControllerRef[0].CanControllerBaseAddress;
-
-	for (ControllerIndex = 0; ControllerIndex < NO_OF_CONTROLLERS_IN_HW;
-			ControllerIndex++) {
-		switch (ControllerIndex) {
-		case 0:
-			ui32Base = CAN0_BASE;
-			break;
-		case 1:
-			ui32Base = CAN1_BASE;
-			break;
-		}
+void Can_MainFunction_Mode(void)
+{
+    /* counter to loop on controllers         */
+	uint8 ControllerIndex  = 0 ;
+	/* static variable to save old state      */
+	static uint8 Old_State = 0 ;
```

Comment:
We should have old_state for each controller.

Reply:
Yes , I'll fix it .